src/java.desktop/share/classes/java/awt/color/ICC_Profile.java

```diff
@@ -1245,25 +1231,12 @@ public void write(OutputStream s) throws IOException {
     }
 
 
-    static byte[] getData(Profile p, int tagSignature) {
-    int tagSize;
-    byte[] tagData;
-
+    private static byte[] getData(Profile p, int tagSignature) {
         try {
-            PCMM mdl = CMSManager.getModule();
-
-            /* get the number of bytes needed for this tag */
-            tagSize = mdl.getTagSize(p, tagSignature);
-
-            tagData = new byte[tagSize]; /* get an array for the tag */
-
-            /* get the tag's data */
-            mdl.getTagData(p, tagSignature, tagData);
-        } catch(CMMException c) {
-            tagData = null;
+            return CMSManager.getModule().getTagData(p, tagSignature);
```

Comment:
Just one step, the array will be created in the native code under proper synchronization.